### PR TITLE
version history: deduplicate the versions displayed

### PR DESCRIPTION
**Pull request by namyyys**
### Problem

In the version history, a dataset will sometimes have the same version displayed multiple times. 
Capture d’écran 2024-07-16 à 10 55 07

After investigating, i found out that this comes from the join clause between the dataset_versions and datasets_view in [this](https://github.com/MarquezProject/marquez/blob/fcfebc55c80b412e975d3b70e487bd9898f2097f/api/src/main/java/marquez/db/DatasetVersionDao.java#L254-L293) function. The symlinks are included in the datasets_view using the same uuid as the dataset and so are displayed in the same logic as the primary dataset. 

### Solution

- Exclude the symlinks from the result of the query to display the version history
- this will be done by creating a CTE 'dataset_symlinks_names' with all the symlinks name. Then we check that all the datasets name included in the list of version does not have a symlink name (and so is not a symlink)

For the same dataset in the above example we have now this display
Capture d’écran 2024-07-16 à 11 14 47

One-line summary:

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [ ] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

File: api/src/test/java/marquez/DatasetIntegrationTest.java [hunk 1]
```diff
@@ -574,4 +576,58 @@ public void testApp_doesNotShowDeletedDatasetAfterUndeleteNamespace() throws IOE
...
+            .inputs(
+                Collections.singletonList(
+                    LineageEvent.Dataset.builder()
+                        .namespace(NAMESPACE_NAME)
+                        .name(DB_TABLE_NAME)
+                        .facets(datasetFacets)
+                        .build()))
+            .outputs(Collections.emptyList())
+            .build();
+    System.out.println("event is " + lineageEvent);
```
Comment (phixMe) [inline api/src/test/java/marquez/DatasetIntegrationTest.java:623]: Remove log statement here.

**Review by phixMe — COMMENTED** [review]
Thanks for the change, happy to merge after the log is removed!

File: api/src/test/java/marquez/DatasetIntegrationTest.java (hunk 1 above)
Comment (namyyys) [reply api/src/test/java/marquez/DatasetIntegrationTest.java]: i pushed a new version removing the log statement

**Review by phixMe — APPROVED** [review]
Thanks for the fix! 💯 👍

**Pull request merged**